### PR TITLE
retroarch: 1.8.5 -> 1.9.2

### PR DESCRIPTION
**Pull request by thiagokokada**
###### Motivation for this change

ZHF: https://github.com/NixOS/nixpkgs/issues/144627

Our retroarch derivation isn't update for quite some time. This one bumps it and all cores, with the help of a `update.py` script that should make it easier to add/bump cores in the future.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 146295` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>libretro.play</li>
  </ul>
</details>
<details>
  <summary>79 packages built:</summary>
  <ul>
    <li>kodiPackages.libretro-genplus</li>
    <li>kodiPackages.libretro-mgba</li>
    <li>kodiPackages.libretro-snes9x</li>
    <li>libretro.atari800</li>
    <li>libretro.beetle-gba</li>
    <li>libretro.beetle-lynx</li>
    <li>libretro.beetle-ngp</li>
    <li>libretro.beetle-pce-fast</li>
    <li>libretro.beetle-pcfx</li>
    <li>libretro.beetle-psx</li>
    <li>libretro.beetle-psx-hw</li>
    <li>libretro.beetle-saturn</li>
    <li>libretro.beetle-saturn-hw</li>
    <li>libretro.beetle-snes</li>
    <li>libretro.beetle-supergrafx</li>
    <li>libretro.beetle-vb</li>
    <li>libretro.beetle-wswan</li>
    <li>libretro.bluemsx</li>
    <li>libretro.bsnes-mercury</li>
    <li>libretro.citra</li>
    <li>libretro.desmume</li>
    <li>libretro.desmume2015</li>
    <li>libretro.dolphin</li>
    <li>libretro.dosbox</li>
    <li>libretro.eightyone</li>
    <li>libretro.fbalpha2012</li>
    <li>libretro.fbneo</li>
    <li>libretro.fceumm</li>
    <li>libretro.flycast</li>
    <li>libretro.fmsx</li>
    <li>libretro.freeintv</li>
    <li>libretro.gambatte</li>
    <li>libretro.genesis-plus-gx</li>
    <li>libretro.gpsp</li>
    <li>libretro.gw</li>
    <li>libretro.handy</li>
    <li>libretro.hatari</li>
    <li>libretro.mame</li>
    <li>libretro.mame2000</li>
    <li>libretro.mame2003</li>
    <li>libretro.mame2003-plus</li>
    <li>libretro.mame2010</li>
    <li>libretro.mame2015</li>
    <li>libretro.mame2016</li>
    <li>libretro.mesen</li>
    <li>libretro.meteor</li>
    <li>libretro.mgba</li>
    <li>libretro.mupen64plus</li>
    <li>libretro.neocd</li>
    <li>libretro.nestopia</li>
    <li>libretro.np2kai</li>
    <li>libretro.o2em</li>
    <li>libretro.opera</li>
    <li>libretro.parallel-n64</li>
    <li>libretro.pcsx_rearmed</li>
    <li>libretro.picodrive</li>
    <li>libretro.ppsspp</li>
    <li>libretro.prboom</li>
    <li>libretro.prosystem</li>
    <li>libretro.quicknes</li>
    <li>libretro.sameboy</li>
    <li>libretro.scummvm</li>
    <li>libretro.smsplus-gx</li>
    <li>libretro.snes9x</li>
    <li>libretro.snes9x2002</li>
    <li>libretro.snes9x2005</li>
    <li>libretro.snes9x2010</li>
    <li>libretro.stella</li>
    <li>libretro.stella2014</li>
    <li>libretro.tgbdual</li>
    <li>libretro.thepowdertoy</li>
    <li>libretro.tic80</li>
    <li>libretro.vba-m</li>
    <li>libretro.vba-next</li>
    <li>libretro.vecx</li>
    <li>libretro.virtualjaguar</li>
    <li>libretro.yabause</li>
    <li>retroarch</li>
    <li>retroarchBare</li>
  </ul>
</details>

**Comment by thiagokokada** [comment]
Tagging as ZHF since this fixes some build issues (`mame` and `play`).

**Comment by thiagokokada** [comment]
Need some help here:
- The cores appear on the `retroarch` (after doing a `retroarch.override { cores = with libretro; [...] }` but I can't them to start a game (they don't appear on the available cores).
- `retroarch` itself is missing resource files (so no icons, wrong font, etc).

**Comment by aanderse** [comment]
@thiagokokada I recently started working on `retroarch` and was splitting each of the cores out into their own file. Each core can have any number of customizations and I think it makes sense to treat them separately.

I'll post some of my initial work soon for discussion if that sounds good to you.

File: pkgs/misc/emulators/retroarch/cores.nix
Comment (hugosenari) [inline pkgs/misc/emulators/retroarch/cores.nix:468]: **👏 praise:** :medal_military:  :medal_military:  :medal_military:

**Comment by thiagokokada** [comment]
> The cores appear on the retroarch (after doing a retroarch.override { cores = with libretro; [...] }; but I can't them to start a game (they don't appear on the available cores).

The strange thing is, it seems that the cores are doing something (if I change the cores, only the file formats recognized by that core shows). But once I select a content it doesn't show the core to play.

**Comment by thiagokokada** [comment]
> > The cores appear on the retroarch (after doing a retroarch.override { cores = with libretro; [...] }; but I can't them to start a game (they don't appear on the available cores).
> 
> The strange thing is, it seems that the cores are doing something (if I change the cores, only the file formats recognized by that core shows). But once I select a content it doesn't show the core to play.

The issue seems to be the `retroarch` bump. If I use `retroarch` 1.8.5 the cores itself works.

**Comment by thiagokokada** [comment]
> The issue seems to be the `retroarch` bump. If I use `retroarch` 1.8.5 the cores itself works.

`retroarch` 1.9.2 works, so using it right now. This at least brings us to a release of retroarch from this year.

**Comment by risicle** [comment]
Didn't manage a full rebuild due to time constraints, but managed to build a large amount of the `nixpkgs-review` set on nixos aarch64.

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 146295` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>libretro.play</li>
  </ul>
</details>
<details>
  <summary>79 packages built:</summary>
  <ul>
    <li>kodiPackages.libretro-genplus</li>
    <li>kodiPackages.libretro-mgba</li>
    <li>kodiPackages.libretro-snes9x</li>
    <li>libretro.atari800</li>
    <li>libretro.beetle-gba</li>
    <li>libretro.beetle-lynx</li>
    <li>libretro.beetle-ngp</li>
    <li>libretro.beetle-pce-fast</li>
    <li>libretro.beetle-pcfx</li>
    <li>libretro.beetle-psx</li>
    <li>libretro.beetle-psx-hw</li>
    <li>libretro.beetle-saturn</li>
    <li>libretro.beetle-saturn-hw</li>
    <li>libretro.beetle-snes</li>
    <li>libretro.beetle-supergrafx</li>
    <li>libretro.beetle-vb</li>
    <li>libretro.beetle-wswan</li>
    <li>libretro.bluemsx</li>
    <li>libretro.bsnes-mercury</li>
    <li>libretro.citra</li>
    <li>libretro.desmume</li>
    <li>libretro.desmume2015</li>
    <li>libretro.dolphin</li>
    <li>libretro.dosbox</li>
    <li>libretro.eightyone</li>
    <li>libretro.fbalpha2012</li>
    <li>libretro.fbneo</li>
    <li>libretro.fceumm</li>
    <li>libretro.flycast</li>
    <li>libretro.fmsx</li>
    <li>libretro.freeintv</li>
    <li>libretro.gambatte</li>
    <li>libretro.genesis-plus-gx</li>
    <li>libretro.gpsp</li>
    <li>libretro.gw</li>
    <li>libretro.handy</li>
    <li>libretro.hatari</li>
    <li>libretro.mame</li>
    <li>libretro.mame2000</li>
    <li>libretro.mame2003</li>
    <li>libretro.mame2003-plus</li>
    <li>libretro.mame2010</li>
    <li>libretro.mame2015</li>
    <li>libretro.mame2016</li>
    <li>libretro.mesen</li>
    <li>libretro.meteor</li>
    <li>libretro.mgba</li>
    <li>libretro.mupen64plus</li>
    <li>libretro.neocd</li>
    <li>libretro.nestopia</li>
    <li>libretro.np2kai</li>
    <li>libretro.o2em</li>
    <li>libretro.opera</li>
    <li>libretro.parallel-n64</li>
    <li>libretro.pcsx_rearmed</li>
    <li>libretro.picodrive</li>
    <li>libretro.ppsspp</li>
    <li>libretro.prboom</li>
    <li>libretro.prosystem</li>
    <li>libretro.quicknes</li>
    <li>libretro.sameboy</li>
    <li>libretro.scummvm</li>
    <li>libretro.smsplus-gx</li>
    <li>libretro.snes9x</li>
    <li>libretro.snes9x2002</li>
    <li>libretro.snes9x2005</li>
    <li>libretro.snes9x2010</li>
    <li>libretro.stella</li>
    <li>libretro.stella2014</li>
    <li>libretro.tgbdual</li>
    <li>libretro.thepowdertoy</li>
    <li>libretro.tic80</li>
    <li>libretro.vba-m</li>
    <li>libretro.vba-next</li>
    <li>libretro.vecx</li>
    <li>libretro.virtualjaguar</li>
    <li>libretro.yabause</li>
    <li>retroarch</li>
    <li>retroarchBare</li>
  </ul>
</details>

**Comment by thiagokokada** [comment]
Tested with Xbox One Controller (needs to manual configuration, but works afterwards).

Tested emulators:
- bsnes-mercury-accuracy
- fbalpha2012
- fbneo (segfault)
- gambatte
- genesis-plus-gx
- mame2010
- mednafen_gba
- mednafen_pce
- mednafen_psx (needs BIOS, but seems to work fine)
- mednafen_snes
- mednafen_wsan
- mgba
- mupen64-plus-next
- parallei-n64
- pcsx_rearmed
- quicknes
- sms-plus
- snes9x
- snes9x2002
- snes9x2005
- snes9x2010
- vba_next

So I tested many emulators and except for fbneo everything seems to work fine. I think this is ready to be reviewed.

**Comment by nrdxp** [comment]
looks nice, but why not the latest release 1.9.13?

**Comment by thiagokokada** [comment]
> looks nice, but why not the latest release 1.9.13?

https://github.com/NixOS/nixpkgs/pull/146295#issuecomment-970681235
https://github.com/NixOS/nixpkgs/pull/146295#issuecomment-970796192
https://github.com/NixOS/nixpkgs/pull/146295#issuecomment-970859165

Now, this probably can be fixed by patching the source code, but I am not sure where it needs to be patched. So for now I decided to settle to 1.9.2 release to at least have some improvements.

**Comment by aanderse** [comment]
> retroarch,libretro: add myself as maintainer

Can we spark up a `libretro` team of maintainers? I'd join that.

**Comment by thiagokokada** [comment]
> > retroarch,libretro: add myself as maintainer
> 
> Can we spark up a `libretro` team of maintainers? I'd join that.

Sure. Can you create an issue?

File: pkgs/misc/emulators/retroarch/cores.nix
Comment (lucasew) [inline pkgs/misc/emulators/retroarch/cores.nix:443]: I think it would be better to create the license then reference that variable here

File: pkgs/misc/emulators/retroarch/cores.nix
Comment (lucasew) [inline pkgs/misc/emulators/retroarch/cores.nix:451]: Same license issue

**Review by lucasew — COMMENTED** [review]
Nice work there!

File: pkgs/misc/emulators/retroarch/cores.nix
Comment (thiagokokada) [reply pkgs/misc/emulators/retroarch/cores.nix:443]: This is also how other cores not affected by this PR is using it: https://github.com/NixOS/nixpkgs/blob/master/pkgs/misc/emulators/retroarch/cores.nix#L808

Also, AFAIK `lib.licenses` only includes open-source licenses. So MAME license is a no-go.

Comment (thiagokokada) [reply pkgs/misc/emulators/retroarch/cores.nix:443]: Correction: `lib.licenses` only includes licenses from SPDX, and MAME is not in SPDX list: https://spdx.org/licenses/

**Review by ratsclub — APPROVED** [review]
It worked correctly here! Thank you for your contribution!

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>libretro.play</li>
  </ul>
</details>
<details>
  <summary>79 packages built:</summary>
  <ul>
    <li>kodiPackages.libretro-genplus</li>
    <li>kodiPackages.libretro-mgba</li>
    <li>kodiPackages.libretro-snes9x</li>
    <li>libretro.atari800</li>
    <li>libretro.beetle-gba</li>
    <li>libretro.beetle-lynx</li>
    <li>libretro.beetle-ngp</li>
    <li>libretro.beetle-pce-fast</li>
    <li>libretro.beetle-pcfx</li>
    <li>libretro.beetle-psx</li>
    <li>libretro.beetle-psx-hw</li>
    <li>libretro.beetle-saturn</li>
    <li>libretro.beetle-saturn-hw</li>
    <li>libretro.beetle-snes</li>
    <li>libretro.beetle-supergrafx</li>
    <li>libretro.beetle-vb</li>
    <li>libretro.beetle-wswan</li>
    <li>libretro.bluemsx</li>
    <li>libretro.bsnes-mercury</li>
    <li>libretro.citra</li>
    <li>libretro.desmume</li>
    <li>libretro.desmume2015</li>
    <li>libretro.dolphin</li>
    <li>libretro.dosbox</li>
    <li>libretro.eightyone</li>
    <li>libretro.fbalpha2012</li>
    <li>libretro.fbneo</li>
    <li>libretro.fceumm</li>
    <li>libretro.flycast</li>
    <li>libretro.fmsx</li>
    <li>libretro.freeintv</li>
    <li>libretro.gambatte</li>
    <li>libretro.genesis-plus-gx</li>
    <li>libretro.gpsp</li>
    <li>libretro.gw</li>
    <li>libretro.handy</li>
    <li>libretro.hatari</li>
    <li>libretro.mame</li>
    <li>libretro.mame2000</li>
    <li>libretro.mame2003</li>
    <li>libretro.mame2003-plus</li>
    <li>libretro.mame2010</li>
    <li>libretro.mame2015</li>
    <li>libretro.mame2016</li>
    <li>libretro.mesen</li>
    <li>libretro.meteor</li>
    <li>libretro.mgba</li>
    <li>libretro.mupen64plus</li>
    <li>libretro.neocd</li>
    <li>libretro.nestopia</li>
    <li>libretro.np2kai</li>
    <li>libretro.o2em</li>
    <li>libretro.opera</li>
    <li>libretro.parallel-n64</li>
    <li>libretro.pcsx_rearmed</li>
    <li>libretro.picodrive</li>
    <li>libretro.ppsspp</li>
    <li>libretro.prboom</li>
    <li>libretro.prosystem</li>
    <li>libretro.quicknes</li>
    <li>libretro.sameboy</li>
    <li>libretro.scummvm</li>
    <li>libretro.smsplus-gx</li>
    <li>libretro.snes9x</li>
    <li>libretro.snes9x2002</li>
    <li>libretro.snes9x2005</li>
    <li>libretro.snes9x2010</li>
    <li>libretro.stella</li>
    <li>libretro.stella2014</li>
    <li>libretro.tgbdual</li>
    <li>libretro.thepowdertoy</li>
    <li>libretro.tic80</li>
    <li>libretro.vba-m</li>
    <li>libretro.vba-next</li>
    <li>libretro.vecx</li>
    <li>libretro.virtualjaguar</li>
    <li>libretro.yabause</li>
    <li>retroarch</li>
    <li>retroarchBare</li>
  </ul>
</details>

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 146295` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>80 packages built:</summary>
  <ul>
    <li>kodiPackages.libretro-genplus</li>
    <li>kodiPackages.libretro-mgba</li>
    <li>kodiPackages.libretro-snes9x</li>
    <li>libretro.atari800</li>
    <li>libretro.beetle-gba</li>
    <li>libretro.beetle-lynx</li>
    <li>libretro.beetle-ngp</li>
    <li>libretro.beetle-pce-fast</li>
    <li>libretro.beetle-pcfx</li>
    <li>libretro.beetle-psx</li>
    <li>libretro.beetle-psx-hw</li>
    <li>libretro.beetle-saturn</li>
    <li>libretro.beetle-saturn-hw</li>
    <li>libretro.beetle-snes</li>
    <li>libretro.beetle-supergrafx</li>
    <li>libretro.beetle-vb</li>
    <li>libretro.beetle-wswan</li>
    <li>libretro.bluemsx</li>
    <li>libretro.bsnes-mercury</li>
    <li>libretro.citra</li>
    <li>libretro.desmume</li>
    <li>libretro.desmume2015</li>
    <li>libretro.dolphin</li>
    <li>libretro.dosbox</li>
    <li>libretro.eightyone</li>
    <li>libretro.fbalpha2012</li>
    <li>libretro.fbneo</li>
    <li>libretro.fceumm</li>
    <li>libretro.flycast</li>
    <li>libretro.fmsx</li>
    <li>libretro.freeintv</li>
    <li>libretro.gambatte</li>
    <li>libretro.genesis-plus-gx</li>
    <li>libretro.gpsp</li>
    <li>libretro.gw</li>
    <li>libretro.handy</li>
    <li>libretro.hatari</li>
    <li>libretro.mame</li>
    <li>libretro.mame2000</li>
    <li>libretro.mame2003</li>
    <li>libretro.mame2003-plus</li>
    <li>libretro.mame2010</li>
    <li>libretro.mame2015</li>
    <li>libretro.mame2016</li>
    <li>libretro.mesen</li>
    <li>libretro.meteor</li>
    <li>libretro.mgba</li>
    <li>libretro.mupen64plus</li>
    <li>libretro.neocd</li>
    <li>libretro.nestopia</li>
    <li>libretro.np2kai</li>
    <li>libretro.o2em</li>
    <li>libretro.opera</li>
    <li>libretro.parallel-n64</li>
    <li>libretro.pcsx_rearmed</li>
    <li>libretro.picodrive</li>
    <li>libretro.play</li>
    <li>libretro.ppsspp</li>
    <li>libretro.prboom</li>
    <li>libretro.prosystem</li>
    <li>libretro.quicknes</li>
    <li>libretro.sameboy</li>
    <li>libretro.scummvm</li>
    <li>libretro.smsplus-gx</li>
    <li>libretro.snes9x</li>
    <li>libretro.snes9x2002</li>
    <li>libretro.snes9x2005</li>
    <li>libretro.snes9x2010</li>
    <li>libretro.stella</li>
    <li>libretro.stella2014</li>
    <li>libretro.tgbdual</li>
    <li>libretro.thepowdertoy</li>
    <li>libretro.tic80</li>
    <li>libretro.vba-m</li>
    <li>libretro.vba-next</li>
    <li>libretro.vecx</li>
    <li>libretro.virtualjaguar</li>
    <li>libretro.yabause</li>
    <li>retroarch</li>
    <li>retroarchBare</li>
  </ul>
</details>

**Comment by thiagokokada** [comment]
Got `play` to build. Tried to run a game and it simply locked up my desktop, but maybe someone else will have more luck.

**Comment by thiagokokada** [comment]
> Got `play` to build. Tried to run a game and it simply locked up my desktop, but maybe someone else will have more luck.

Should fix this issue: https://github.com/NixOS/nixpkgs/issues/108988

**Comment by thiagokokada** [comment]
@aanderse Can I get your review here?

**Review by aanderse — APPROVED** [review]
This is great! Thank you for your hard work on this! We should coordinate what happens after we merge here. Are you ever on the NixOS matrix chat?

**Comment by thiagokokada** [comment]
Ok, thanks, merging it.

> This is great! Thank you for your hard work on this! We should coordinate what happens after we merge here. Are you ever on the NixOS matrix chat?

I am, not really active though. But you can send me a DM. Nick is `@k0kada:matrix.org`.

**Pull request merged**
